luminance-webgl/src/webgl2/state.rs

```diff
@@ -800,6 +810,24 @@ fn get_ctx_face_culling_mode(
   }
 }
 
+fn load_webgl_extensions(ctx: &mut WebGl2RenderingContext) -> Result<(), StateQueryError> {
+  let required_extensions = vec!["OES_texture_float_linear", "EXT_color_buffer_float"];
+
+  let missing_extensions: Vec<String> = required_extensions
+    .into_iter()
+    .map(|ext| (ext, ctx.get_extension(ext)))
+    .filter(|(_, object)| object.is_err() || object.as_ref().unwrap().is_none())
```

Comment:
I’m not a huge fan of this `is_err() || unwrap` pattern. You can easily rewrite that with:

```suggestion
    .filter(|(_, object)| object.as_ref().ok().map(Option::is_none).unwrap_or(true))
```

Reply:
I split it into 2 steps:
- determine the available extensions.  which is very nice with `result.ok().flatten().map(|_| ext)`
- length check, then filter down to the missing extensions.

Much cleaner now!